lib: nrf_modem_lib: rewrite UART trace backend to use zephyr driver (#10891)
nrfconnect/sdk-nrf: doc/nrf/working_with_nrf/nrf91/nrf9160.rst

      The device resets and runs the programmed sample or application.

.. _nrf91_modem_trace_uart_snippet:

nRF91 modem tracing with UART backend using snippets
====================================================

> The :ref:`snippets` is tailored for tracing on the nRF91 Series devices but can work with other boards as well.

Reviewer: ```suggestion
:ref:`snippets` are tailored for tracing on the nRF91 Series devices but can work with other boards as well.
```

It enables the :kconfig:option:`CONFIG_NRF_MODEM_LIB_TRACE` Kconfig option and chooses the Zephyr UART driver for the backend, with the necessary Kconfig options.

> Snippets also enables the UART1 peripheral with a baud rate of 1 Mbd and hardware flow control enabled.

Reviewer 2: ```suggestion
Snippets also enable the UART1 peripheral with a baud rate of 1 Mbd and hardware flow control enabled.
```

> If this configuration does not match your requirements, you can add a snippet or Kconfig and device tree overlays to your application with the desired setup.

Reviewer: ```suggestion
If this configuration does not match your requirements, you can add a snippet or Kconfig and devicetree overlays to your application with the desired setup.
```

      The device resets and runs the programmed sample or application.

.. _nrf91_modem_trace_uart_snippet:

nRF91 modem tracing with UART backend using snippets
====================================================

> The |NCS| provides built-in snippets that can be used with our sample and applications.
> See the zephyr's :ref:`snippets` documentation to know more about snippets in general.
> 

Reviewer: ```suggestion
```
The link to zephyr documentation is provided in line no:274.

> The snippet is tailored for tracing on the nRF91 Series devices but can work with other boards as well.

Reviewer: ```suggestion
The :ref:`snippets` is tailored for tracing on the nRF91 Series devices but can work with other boards as well.
```
